sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_patch.py

```diff
@@ -0,0 +1,694 @@
+        http_request_kwargs = get_http_request_kwargs(kwargs)
+        # ignoring return type because the generated client operations are not annotated w/ cls return type
+        schema_properties: Dict[
+            str, Union[int, str]
+        ] = self._generated_client._get_schema_properties_by_content(  # pylint:disable=protected-access
+            group_name=group_name,
+            schema_name=name,
+            schema_content=cast(IO[Any], definition),
+            content_type=kwargs.pop("content_type", get_content_type(format)),
+            cls=partial(prepare_schema_properties_result, format),
+            **http_request_kwargs,
+        )
+        properties = cast("SchemaPropertiesDict", schema_properties)
+        return SchemaProperties(**properties)
+
+
+class SchemaProperties:
+    """
+    Meta properties of a schema.
+
+    :ivar id: References specific schema in registry namespace.
+    :vartype id: str
+    :ivar format: Format for the schema being stored.
+    :vartype format: ~azure.schemaregistry.SchemaFormat
+    :ivar group_name: Schema group under which schema is stored.
+    :vartype group_name: str
+    :ivar name: Name of schema.
+    :vartype name: str
+    :ivar version: Version of schema.
+    :vartype version: int
+    """
+
+    def __init__(self, **kwargs: Any) -> None:
+        self.id = kwargs.pop("id")
+        self.format = kwargs.pop("format")
+        self.group_name = kwargs.pop("group_name")
+        self.name = kwargs.pop("name")
+        self.version = kwargs.pop("version")
+
+    def __repr__(self):
+        return (
+            f"SchemaProperties(id={self.id}, format={self.format}, "
+            f"group_name={self.group_name}, name={self.name}, version={self.version})"[:1024]
+        )
+
+
+class Schema:
+    """
+    The schema content of a schema, along with id and meta properties.
+
+    :ivar definition: The content of the schema.
+    :vartype definition: str
+    :ivar properties: The properties of the schema.
+    :vartype properties: SchemaProperties
+    """
+
+    def __init__(self, **kwargs: Any) -> None:
+        self.definition = kwargs.pop("definition")
+        self.properties = kwargs.pop("properties")
+
+    def __repr__(self):
+        return f"Schema(definition={self.definition}, properties={self.properties})"[:1024]
+
+
+###### Encoder Protocols ######
+
+
+class SchemaContentValidate(Protocol):
+    # TODO: make __call__ a public API so that docs show, until then, keep below docstring
+
+    """
+    Callable protocol which validates content against provided schema. If invalid, raises Exception.
+     Else, returns None.
+
+    :param mapping[str, any] schema: The schema to validate against.
+    :param mapping[str, any] content: The content to validate.
+
+    :rtype: None
+    :returns: None if valid.
+    :raises: Exception if content is invalid against provided schema.
+    """
+
+    def __call__(self, schema: Mapping[str, Any], content: Mapping[str, Any]) -> None:
```

Comment:
Where are we doing the validation?

Reply:
We call the callback that follows this protocol in JsonSchemaEncoder.encode/decode here: https://github.com/Azure/azure-sdk-for-python/blob/5513d6981f51196024fe2edaa9470712e4f35d65/sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/encoder/jsonencoder/_utils.py#L103.